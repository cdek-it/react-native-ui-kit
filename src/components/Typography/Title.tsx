import { Text, type TextProps } from 'react-native'

import { makeStyles } from '../../utils/makeStyles'

export interface TitleProps extends TextProps {
  readonly level: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
}

export const Title = ({ level, style, ...other }: TitleProps) => {
  const styles = useStyles()

  return <Text style={[styles.text, styles[level], style]} {...other} />
}

const useStyles = makeStyles(({ theme }) => ({
  text: {
    color: theme.General.textColor,
    fontWeight: 700,
    includeFontPadding: false,
    verticalAlign: 'middle',
  },
  h1: { fontSize: 28 },
  h2: { fontSize: 21 },
  h3: { fontSize: 17.5 },
  h4: { fontSize: 15.75 },
  h5: { fontSize: 14 },
  h6: { fontSize: 12.25 },
}))
