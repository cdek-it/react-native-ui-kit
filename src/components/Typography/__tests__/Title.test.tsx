import { render } from '@testing-library/react-native'

import { Title, type TitleProps } from '../Title'

describe('Title component tests', () => {
  const snapshotCases = generatePropsCombinations<TitleProps>({
    level: ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'],
  })

  test.each(snapshotCases)('level = $level', (props) => {
    const renderedTitle = render(<Title {...props}>Text</Title>)

    expect(renderedTitle.toJSON()).toMatchSnapshot()
  })
})
