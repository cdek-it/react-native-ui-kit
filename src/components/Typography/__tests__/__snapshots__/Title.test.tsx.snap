// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component tests level = h1 1`] = `
<Text
  style={
    [
      {
        "color": "rgba(0, 0, 0, 0.8000)",
        "fontWeight": 700,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      },
      {
        "fontSize": 28,
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`Title component tests level = h2 1`] = `
<Text
  style={
    [
      {
        "color": "rgba(0, 0, 0, 0.8000)",
        "fontWeight": 700,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      },
      {
        "fontSize": 21,
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`Title component tests level = h3 1`] = `
<Text
  style={
    [
      {
        "color": "rgba(0, 0, 0, 0.8000)",
        "fontWeight": 700,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      },
      {
        "fontSize": 17.5,
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`Title component tests level = h4 1`] = `
<Text
  style={
    [
      {
        "color": "rgba(0, 0, 0, 0.8000)",
        "fontWeight": 700,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      },
      {
        "fontSize": 15.75,
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`Title component tests level = h5 1`] = `
<Text
  style={
    [
      {
        "color": "rgba(0, 0, 0, 0.8000)",
        "fontWeight": 700,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      },
      {
        "fontSize": 14,
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`Title component tests level = h6 1`] = `
<Text
  style={
    [
      {
        "color": "rgba(0, 0, 0, 0.8000)",
        "fontWeight": 700,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      },
      {
        "fontSize": 12.25,
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;
